lua/lspconfig/server_configurations/circom-lsp.lua

```diff
@@ -0,0 +1,26 @@
+local util = require 'lspconfig.util'
+
+local root_files = {
+  'package.json',
+  'tsconfig.json',
+  'pyproject.toml',
+  'Cargo.toml',
+}
+
+return {
+  default_config = {
+    cmd = { 'circom-lsp' },
+    filetypes = { 'circom' },
+    root_dir = function()
+      local root = util.root_pattern(root_files)(vim.fn.expand '%:p')
+      return root or vim.api.nvim_buf_get_name(0)
```

Comment:
i don't think use current bufname is useful if you want root is current dir just `single_file_support = true`

Reply:
Yes I agree 
Simplify the configuration by setting single_file_support to true. This approach is more straightforward, as you don't need to specify root_files or create a custom root_dir function.

Reply:
Set single_file_support to true for improved simplicity.